### PR TITLE
First Try at GANs

### PR DESCRIPTION
**Pull request by BaladityaY**
Cleaned up gradient_ascent.py and implemented GAN training based off pretrained squeezenet in gan_gradient_ascent.py

### CODE REVIEW

File: visualizations/gradient_ascent.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
-import numpy as np
-
 import torch
 from torch.autograd import Variable
+<<<<<<< HEAD
```
Comment (tpankaj) [inline visualizations/gradient_ascent.py]: Fix merge conflict

File: visualizations/gradient_ascent.py [hunk 2]
```diff
@@ -34,6 +37,115 @@
 
 all_layer_funcs.append(lambda x: x.view(x.size(0), -1))
 
+<<<<<<< HEAD
```
Comment (tpankaj) [inline visualizations/gradient_ascent.py]: Fix merge conflict

**Review by tpankaj — CHANGES_REQUESTED** [review]
Please fix all merge conflict issues that I commented on, and also use the automated code review feedback to fix your unused imports and other issues.

File: visualizations/gan_gradient_ascent.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+loss_record_g['train'] = Utils.Loss_Record()
+loss_record_g['val'] = Utils.Loss_Record()
+
+rate_counter_d = Utils.Rate_Counter()
+
+rate_counter_g = Utils.Rate_Counter() 
+
+data = Data.Data()
+
+'''
```
Comment (tpankaj) [inline visualizations/gan_gradient_ascent.py]: Do you still need this commented stuff?

File: visualizations/gan_gradient_ascent.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+                net_g.zero_grad()
+                loss_g.backward()
+                nnutils.clip_grad_norm(net_g.parameters(), 1.0)
+                optimizer_g.step()
+
+            loss_record_g[mode].add(loss_g.data[0])
+            rate_counter_g.step()
+            
+            '''
+            if save_timer.check():
```
Comment (tpankaj) [inline visualizations/gan_gradient_ascent.py]: Do you need this commented part?

**Review by tpankaj — DISMISSED** [review]
Please fix the Codacy errors before we merge.

**Review by raunakdoesdev — DISMISSED** [review]
Looks good. I think we can merge this in.

**Review by raunakdoesdev — APPROVED** [review]
Looks good. I think we can merge this in.

**Comment by raunakdoesdev** [comment]
Although this looks good I would like for the ARGS to be imported inside outside of a if __name__ == 'main'. Since sometimes for testing purposes we like to run the network test by itself using the unit test.I merged this now, since overall it looked good but it would be nice if we could do this in a future PR.

**Pull request merged**